### PR TITLE
Add URL validation #97

### PR DESCRIPTION
**Pull request by u5surf**
#97 
I added url validation following rules of Go’s net/url Parse.

### CODE REVIEW

**Comment by Kashomon** [comment]
Thanks for the PR!

File: pkg/validation/validate_url_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			err := validateURL(tc.in)
+			if err == nil {
+				if tc.errSubstr != "" && !strings.Contains(err.Error(), tc.errSubstr) {
+					t.Errorf("got nil error, but expected one with %q", tc.errSubstr)
+					return
+				}
+				return // no error, not a problem
+			} else {
```
Comment (Kashomon) [inline pkg/validation/validate_url_test.go]: nit: don't need the else here.

**Comment by Kashomon** [comment]
Can you update the bazel BUILD file? Once you have `bazel` [installed](https://docs.bazel.build/versions/master/install.html), you can run `bazel run //:gazelle`. Or, I suppose you can just run `gazelle` manually: https://github.com/bazelbuild/bazel-gazelle#running-gazelle-with-go

**Comment by u5surf** [comment]
> Can you update the bazel BUILD file? Once you have `bazel` [installed](https://docs.bazel.build/versions/master/install.html), you can run `bazel run //:gazelle`. Or, I suppose you can just run `gazelle` manually: https://github.com/bazelbuild/bazel-gazelle#running-gazelle-with-go

```
k8s-cluster-bundle $ bazel run //:gazelle
INFO: Analysed target //:gazelle (38 packages loaded, 7008 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle.bash
INFO: Elapsed time: 108.618s, Critical Path: 64.03s, Remote (0.00% of the time): [queue: 0.00%, setup: 0.00%, process: 0.00%]
INFO: 37 processes: 37 processwrapper-sandbox.
INFO: Build completed successfully, 50 total actions
INFO: Build completed successfully, 50 total actions
```

File: pkg/validation/validate_url_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			desc: "success: no scheme",
+			in:   "foo/bar/biff",
+			path: "foo/bar/biff",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			err := validateURL(tc.in)
+			if err == nil {
+				if tc.errSubstr != "" && !strings.Contains(err.Error(), tc.errSubstr) {
```
Comment (Kashomon) [inline pkg/validation/validate_url_test.go]: this isn't quite right. err == nil means that err.Error() is dereferencing a nil object.

There should be a case where err != nil and you check this. I got this wrong in validate_names, which is probably why this is wrong here. I'll go fix that.

**Pull request merged**